src/data/userDAO.ts

```diff
@@ -0,0 +1,83 @@
+import AWS, { DynamoDB } from 'aws-sdk';
+import fetch from 'node-fetch';
+
+const TABLE_NAME = 'users';
+
+export type DiscordUser = {
+    id: string;
+    username: string;
+};
+
+export type User = {
+    id: string;
+    username: string;
+    accessToken: string;
+    refreshToken: string;
+};
+
+export class UserDAO {
+    private dynamodb: DynamoDB;
+
+    constructor() {
+        this.dynamodb = new AWS.DynamoDB();
+    }
+
+    public async saveUser(accessToken: string, refreshToken: string): Promise<any> {
+        const user = await this.getDiscordUser(accessToken);
+        return new Promise((resolve, reject) => {
+            const params = {
+                Item: {
+                    id: { S: user.id },
+                    username: { S: user.username },
+                    accessToken: { S: accessToken },
+                    refreshToken: { S: refreshToken }
+                },
+                TableName: TABLE_NAME
+            };
+            this.dynamodb.putItem(params, (err) => {
+                if (err) {
+                    console.log(err, err.stack);
+                    reject(err);
+                }
+                resolve();
+            });
+        });
+    }
+
+    public async getUserFromToken(token: string): Promise<User> {
+        const user = await this.getDiscordUser(token);
+        return await this.getUser(user.id);
```

Comment:
No need to await when you are returning, **unless** you want to catch the error in this functions scope (using try/catch). A good write up is here https://jakearchibald.com/2017/await-vs-return-vs-return-await/